### PR TITLE
[IMP] sale: docstrings improvements/fixes

### PR DESCRIPTION
**Pull request by Feyensv**
---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale/models/sale_order.py [hunk 1]
```diff
@@ -748,9 +748,17 @@ def _find_mail_template(self):
             return self._get_confirmation_template()
 
     def _get_confirmation_template(self):
+        """ Get the mail template sent on SO confirmation (or for confirmed SO's)
```
Comment (Demesmaeker) [inline addons/sale/models/sale_order.py:751]:
```suggestion
        """ Get the mail template sent on SO confirmation (or for confirmed SO's).
```
=P

File: addons/sale/models/sale_order.py [hunk 2]
```diff
@@ -748,9 +748,17 @@ def _find_mail_template(self):
...
+
+        :return: `mail.template` record or None if default template wasn't found
+        """
         return self.env.ref('sale.mail_template_sale_confirmation', raise_if_not_found=False)
 
     def action_quotation_sent(self):
+        """ Mark the given draft quotations as sent.
```
Comment (Demesmaeker) [inline addons/sale/models/sale_order.py:758]:
```suggestion
        """ Mark the given draft quotation(s) as sent.
```
You sometimes put parenthesis, sometimes you don't. Maybe uniformize all of it? :man_shrugging:

File: addons/sale/models/sale_order.py [hunk 3]
```diff
@@ -760,6 +768,14 @@ def action_quotation_sent(self):
         self.write({'state': 'sent'})
 
     def action_confirm(self):
+        """ Confirm the given quotation(s) and sets their confirmation date.
```
Comment (Demesmaeker) [inline addons/sale/models/sale_order.py:771]:
```suggestion
        """ Confirm the given quotation(s) and set their confirmation date.
```

File: addons/sale/models/sale_order.py [hunk 4]
```diff
@@ -1361,56 +1392,64 @@ def _create_upsell_activity(self):
...
+        """ Prepare SO analytic account creation values.
 
-        :param prefix: The prefix of the to-be-created analytic account name
-        :type prefix: string
-        :return: dictionary of value for new analytic account creation
+        :param str prefix: The prefix of the to-be-created analytic account name
+        :return: `account.analytic.account` creation values
+        :rtype: dict
         """
+        self.ensure_one()
```
Comment (Demesmaeker) [inline addons/sale/models/sale_order.py:1401]: Don't we `Note: self.ensure_one()` like in `payment`?

File: addons/sale/models/sale_order.py [hunk 5]
```diff
@@ -1361,56 +1392,64 @@ def _create_upsell_activity(self):
...
             'plan_id': plan.id,
-            'partner_id': self.partner_id.id
+            'partner_id': self.partner_id.id,
         }
 
     def _create_analytic_account(self, prefix=None):
+        """ Create a new analytic account for the given orders.
+
+        :param str prefix: if specified, the account name will be '<prefix>: <so_reference>'.
+          if not, the account name will be the Sales Order reference.
```
Comment (Demesmaeker) [inline addons/sale/models/sale_order.py:1425]:
```suggestion
        :param str prefix: if specified, the account name will be '<prefix>: <so_reference>'.
            If not, the account name will be the Sales Order reference.
```

**Review by Demesmaeker — APPROVED** [review]
:duck:

File: addons/sale/models/sale_order.py (hunk 4 above)
Comment (Feyensv) [reply addons/sale/models/sale_order.py:1401]: not a fan of specifying it in the docstring personally :D

**Comment by Feyensv** [comment]
@robodoo r+

**Pull request closed without merging**